src/codegen.cpp

```diff
@@ -6462,6 +6471,12 @@ static std::pair<std::unique_ptr<Module>, jl_llvm_functions_t>
         jl_sym_t *s = (jl_sym_t*)jl_array_ptr_ref(src->slotnames, i);
         jl_value_t *argType = jl_nth_slot_type(lam->specTypes, i);
         bool isboxed = deserves_argbox(argType);
+        if (i == 0 && ctx.is_opaque_closure) {
+            // The opaque closure itself is passed as the first argument, but
+            // specTypes stores the closure env type.
+            argType = jl_unwrap_unionall(
+                (jl_value_t*)jl_opaque_closure_type);
```

Comment:
seems strange to intentionally introduce an invalid type object here as the argument type with `jl_unwrap_unionall`

Reply:
Yeah, this is to make `emit_getfield_knownidx` below work since the layout is independent of the type parameters. I guess I could just do `OpaqueClosure{Tuple{Any}, Any}` to make something valid or we could just manually compute the offset. Maybe the latter makes the most sense.

Reply:
ah, yeah, I'm surprised we don't end up tagging that load with some nonsensical type too and causing confusion

I guess the latter possibly does make the most sense, as later the intent is to inline the data, rather than making it a separate Tuple{Any} allocation

Reply:
Yeah, alright, let me just do that.